modules/dnn/src/op_webnn.cpp

```diff
@@ -0,0 +1,136 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018-2019, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+#include <fstream>
+#include "op_webnn.hpp"
+
+#include <opencv2/core/utils/configuration.private.hpp>
+#include <opencv2/core/utils/logger.hpp>
+
+#include "opencv2/core/utils/filesystem.hpp"
+#include "opencv2/core/utils/filesystem.private.hpp"
+
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace cv { namespace dnn {
+
+#ifdef HAVE_WEBNN
+
+template<typename T>
+static inline std::vector<T> getShape(const Mat& mat)
+{
+    std::vector<T> result(mat.dims);
+    for (int i = 0; i < mat.dims; i++)
+        result[i] = (T)mat.size[i];
+    return result;
+}
+
+// WebnnGraph
+
+WebnnGraph::WebnnGraph()
+{
+
+}
+
+bool WebnnGraph::isInitialized()
+{
+
+}
+
+void WebnnGraph::init(Target targetId)
+{
+
+}
+
+void WebnnGraph::forward(const std::vector<Ptr<BackendWrapper> >& outBlobsWrappers)
+{
+
+}
+
+void WebnnGraph::createGraph(Target targetId)
+{
+
+}
+
+// WebnnBackendNode
+
+WebnnBackendNode::WebnnBackendNode(const std::vector<Ptr<BackendNode> >& nodes,
+                                         Ptr<Layer>& cvLayer_, std::vector<Mat*>& inputs,
+                                         std::vector<Mat>& outputs, std::vector<Mat>& internals)
+    : BackendNode(DNN_BACKEND_WEBNN))
+{
+    
+}
+
+void WebnnBackendNode::setName(const std::string& name)
+{
+
+}
+
+// WebnnBackendWrapper
+
+WebnnBackendWrapper::WebnnBackendWrapper(const Mat& m)
+{
+    std::vector<uint32_t> shape = getShape<uint32_t>(m);
+    if (m.type() == CV_16F)
+    {
+        buffer = (float16_t*) m.data;
```

Comment:
If `m` released, the buffer will be an invalid pointer. So please allocate the memory and use `buffer` to point it. `buffer` itself could be a smart pointer, so the allocated memory can be freed when `WebnnBackendWrapper` is released. Please refer to similar usage of webnn-native: https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn_native/dml/GraphDML.h#L97, e.g., make `buffer` to be a `unique_ptr<char> buffer`.

https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn_native/dml/GraphDML.cpp#L480-L481